src/utils/temp_folder.test.ts

```diff
@@ -0,0 +1,44 @@
+import { expect } from 'chai';
+import { cleanUpTempFolder, getTempFolder } from './temp_folder';
+import * as fs from 'fs';
+import * as os from 'os';
+
+describe('temp folder functions', () => {
+	describe('getTempFolder function', () => {
+		it('returns a folder that exists', () => {
+			const tempFolderPath = getTempFolder();
+			expect(fs.existsSync(tempFolderPath)).to.equal(true);
+		});
+
+		it('can be called twice in a row', () => {
+			const tempFolderPath = getTempFolder();
+			const tempFolderPath2 = getTempFolder();
+			expect(fs.existsSync(tempFolderPath)).to.equal(true);
+			expect(fs.existsSync(tempFolderPath2)).to.equal(true);
+		});
+
+		it('returns a folder that contains the correct subfolders', () => {
+			const tempFolderPath = getTempFolder();
+			const expectedPathComponent = os.platform() === 'win32' ? '\\ardrive-temp' : '/.ardrive/temp';
```

Comment:
We usually avoid dynamic behaviors in the unit tests. In this case we are making use of `os.plarform` which is not a constant value. We could instead mock this all.

Reply:
That's actually a pretty good idea. We could make the os string injectable and have it use os.platform as the default value if not specified. I would say non-blocking right now.

Reply:
Not a blocker +1